src/wp-includes/rest-api/endpoints/class-wp-rest-themes-controller.php

```diff
@@ -115,6 +115,28 @@ public function prepare_item_for_response( $theme, $request ) {
 		$data   = array();
 		$fields = $this->get_fields_for_response( $request );
 
+		$field_mappings    = array(
+			'author'      => 'Author',
+			'author_name' => 'Author Name',
+			'author_uri'  => 'Author URI',
+			'description' => 'Description',
+			'name'        => 'Name',
+			'stylesheet'  => 'Stylesheet',
+			'template'    => 'Template',
+			'theme_uri'   => 'Theme URI',
+			'version'     => 'Version',
+		);
+		$fields_to_include = array_intersect( array_keys( $field_mappings ), $fields );
+
+		foreach ( $fields_to_include as $field ) {
+			$data[ $field ] = $theme[ $field_mappings[ $field ] ];
```

Comment:
This ends up giving us a mixture of translated/untranslated and markedup/unmarked up properties.

I'd prefer it if we followed the `raw/rendered` format of the other REST API endpoints. So that would look something more like this.

```php
$data['author'] = array(
    'raw' => $theme->display('Author', false, true ),
    'rendered' => $theme->display('Author' ),
);
```

We'd follow that pattern for `Author`, `Author URI`, `Description`, `Name`, and `Theme URI`. And we would drop the separate `Author Name` field entirely.

Reply:
Good point, thanks!

This works great for `author`; e.g. for the default theme:

```js
{
	raw: 'Michael Heilemann',
	rendered: '<a href="http://binarybonsai.com/">Michael Heilemann</a>'
}
```

I'm not so sure about some of the others. E.g. do we really want a `rendered` `AuthorURI`? Should that be `<a href="http://binarybonsai.com/">http://binarybonsai.com/</a>`? IMO, the `rendered` `Author` makes more sense as link text. This raises the question of whether to keep `AutorURI` around as a 'plain' string field, or whether to drop it. (I'm keeping it for now, as it seems useful in a client to build a clickable link from `author_uri` and `raw` `author` without needing to sanitize `rendered` `author`.)

The same argument largely applies to `ThemeURI`, where a clickable link could be built from that field (if left as a plain string) and `Name`.

Furthermore, I'm not sure if `$theme->display()` is working so well for a number of fields; `Description` always seems to contain a link for me (again for the default theme):

```
4) WP_Test_REST_Themes_Controller::test_theme_description
Failed asserting that Array &0 (
    'raw' => 'The default WordPress theme based on the famous <a href="http://binarybonsai.com/kubrick/">Kubrick</a>.'
    'rendered' => 'The default WordPress theme based on the famous <a href="http://binarybonsai.com/kubrick/">Kubrick</a>.'
) is identical to [...]
```

Finally, what about `Name`? Is markup allowed in that field? If no, I'd also stick with a plain string :thinking:

Reply:
Using `display` means we'd be using the existing [`WP_Theme::markup_header`](https://github.com/WordPress/wordpress-develop/blob/86cbde382a0d323480a14b53c86f274599880ee2/src/wp-includes/class-wp-theme.php#L885). We wouldn't be making those urls links, WP applies `esc_url` to those values.

For the description, the `<a>` tag is there because it is present in the [theme's actual description](https://themes.trac.wordpress.org/browser/default/1.7.2/style.css#L4). It isn't be added as an artifact of using `display`. 

> Finally, what about Name? Is markup allowed in that field? If no, I'd also stick with a plain string 🤔

Name get's a fallback set when using `display`.

Reply:
Ah, gotcha. Okay, I'll use `display` for those :+1:

Reply:
8b16987